gnucash/report/trep-engine.scm

```diff
@@ -1411,6 +1484,36 @@ be excluded from periodic reporting.")
                              (cons 'merge-dual-column? #f))))
              '())
 
+         (if (column-uses? 'running-sec)
+             (list (list (cons 'heading (header-commodity (G_ "Secondary Running Subtotal")))
+                         (cons 'calc-fn converted-running-sec)
+                         (cons 'reverse-column? #f)
+                         (cons 'subtotal? #f)
+                         (cons 'start-dual-column? #f)
+                         (cons 'friendly-heading-fn (const ""))
+                         (cons 'merge-dual-column? #f)))
+             '())
+
+         (if (column-uses? 'running-prime)
+             (list (list (cons 'heading (header-commodity (G_ "Running Subtotal")))
+                         (cons 'calc-fn converted-running-prime)
+                         (cons 'reverse-column? #f)
+                         (cons 'subtotal? #f)
+                         (cons 'start-dual-column? #f)
+                         (cons 'friendly-heading-fn (const ""))
+                         (cons 'merge-dual-column? #f)))
+             '())
+
+         (if (column-uses? 'running-grand-total)
+             (list (list (cons 'heading (header-commodity (G_ "Running Total")))
```

Comment:
To be consistent, I'd rename these to "Running Secondary Subtotal" "Running Primary Subtotal", "Running Grand Total".

Reply:
Yes I went back and forth on this and don't love it either way. 'Running Secondary Subtotal" "Running Primary Subtotal", "Running Grand Total" is nice because consistent. However it makes for long headings with unecessary info when not using all 3 . Most use case probably won't have secondary sutotals. Hence you are left with "Running Primary Subtotal", "Running Grand Total". I was just trying to be shorter. Or you just have "Running Grand Total" when these is no other total anyway so it seens redudant and looks a bit weird as a column heading (not really a problem on the last line as a row heading). 

What if I make the labels dynamic depending on what running totals are displayed?
- If only running grand total is showns with no subtotals, then we keep it as "Running Total" otherwise it will say "Running Grand Total" when subtotals are present as well (subtotals can be running or not, so it removes the confusion either way).
- For subtotals, if only primary subtotal exist without secondary, then we keep it as "Running Subtotal" otherwise "Running Primary Subtotal". 

So basically:
Just grand: "Running Total"
grand+primary: "Running Subtotal", "Running Grand Total"
all 3: as you show
primary only: "Running Subtotal"
primary and secondary: "Running Secondary Subtotal" "Running Primary Subtotal"

I don't think it adds a lot of code, just some logic on the grand and primary labels and all "column-uses" constants are already there for the needed logic. It would give us "best of both worlds". What do you think? I'll give it a try, it shouldn't be much to implement -- if it is I'll stop and consider your orignal ask.

Reply:
Just be sure to include comments beginning with "Translators:" explaining what's going on so that the translators&mdash;who can't see the code without extra effort and might not be able to understand it anyway&mdash;have a chance of getting the right meanings in their own languages.

Reply:
See https://github.com/Gnucash/gnucash/pull/1617/commits/2236cd292fff825ac94707d0f999e76ff032e9d0
Much better like this I think. Will squash this commit with main one if we move forward with it

All 3:
![image](https://github.com/Gnucash/gnucash/assets/267163/9142354d-328f-42cd-bdea-ab07c6f05394)

Primary + Grand
![image](https://github.com/Gnucash/gnucash/assets/267163/e0fb32ed-eaa9-4a59-b43b-7262cd796780)

Grand alone
![image](https://github.com/Gnucash/gnucash/assets/267163/57e88f23-dde3-4c39-b87a-39f6b0cdbfe0)

Reply:
@jralls Are these regular comments or some other way to pass them to the translators? Also do the translator see each G_ string (pun intended I guess) separately or in context, together? Then they will see:
Running Total
Running Grand Total
Running Subtotal
Running Primary Subtotal
Running Secondary Subtotal

I suppose the comments would be for Running Total and Running Subtotal:
- Running Total is a shorter version of Running Grand Total used in the context where a Running Grand Total is displayed without subtotals.
- Running Subtotal is a shorter version of Running Primary Subtotal used when a Running Primary Subtotal is displayed without a Secondary Subtotal.

Reply:
Just a regular comment beginning with the word "Translators:", see e.g. [this one](https://github.com/Gnucash/gnucash/blob/6111184c735f98da7b2f5dcf529f5660a16d6bb3/gnucash/report/trep-engine.scm#L1271) and on the [corresponding weblate page](https://hosted.weblate.org/translate/gnucash/gnucash/fr/?q=+source%3A%3DL+&sort_by=-priority%2Cposition&offset=2) you can see that they get just one string at a time. Along with the comment and context* when available there are links to the code in the String Information block, but very few translators are programmers so that seldom does them much good.

\* context in this case is a differentiator to create multiple instances of single words or short phrases that when used in different places might have different translations. In scheme the string is introduced with `(C_ "…")` instead of `(G_ "…")`.

Reply:
See https://github.com/Gnucash/gnucash/pull/1617/commits/e928c693dfc7f7399d01a7672ae2037772eb25ae for translator comments. I will squash this as well but can somebody first check I put these where they should go (and whether they make sense)? Also I assume that if a term is used more than once, I only put the comment on the first instance.

Reply:
The translator comments are fine.